src/main/java/org/broadinstitute/hellbender/tools/spark/sv/evidence/XGBoostEvidenceFilter.java

```diff
@@ -0,0 +1,609 @@
+        final CoverageScaledOverlapInfo clusterOverlapInfo = getClusterOverlapInfo(evidence);
+
+        // calculate properties related to overlap of intervals on the reference genome
+        final double referenceGapOverlap = getGenomeIntervalsOverlap(evidence, genomeGaps);
+        final double umapS100 = getGenomeIntervalsOverlap(evidence, umapS100Mappability);
+
+        return new EvidenceFeatures(
+            new double[]{
+                cigarQualityInfo.basesMatched, cigarQualityInfo.referenceLength, evidenceType, mappingQuality, templateSize,
+                individualOverlapInfo.numOverlap, individualOverlapInfo.overlapMappingQuality,
+                individualOverlapInfo.meanOverlapMappingQuality, individualOverlapInfo.numCoherent,
+                individualOverlapInfo.coherentMappingQuality,
+                clusterOverlapInfo.numOverlap, clusterOverlapInfo.overlapMappingQuality,
+                clusterOverlapInfo.meanOverlapMappingQuality, clusterOverlapInfo.numCoherent,
+                clusterOverlapInfo.coherentMappingQuality,
+                referenceGapOverlap, umapS100
+            }
+        );
+    }
+
+    private int getMappingQuality(final BreakpointEvidence evidence) {
+        // Note: return "max" mapping quality for non-ReadEvidence. Reasoning: some features depend on sum or average of
+        // read qualities. Non-ReadEvidence isn't *bad* per se, so give it a good score.
+        return evidence instanceof BreakpointEvidence.ReadEvidence ?
+            ((BreakpointEvidence.ReadEvidence) evidence).getMappingQuality() : 60;
+    }
+
+    private double getTemplateSize(final BreakpointEvidence evidence) {
+        if(evidence instanceof BreakpointEvidence.ReadEvidence) {
+            // For ReadEvidence, return templateSize as percentile of library's cumulative density function:
+            final String readGroup = ((BreakpointEvidence.ReadEvidence) evidence).getReadGroup();
+            final String library = readGroupToLibraryMap.get(readGroup);
+            final LibraryStatistics libraryStatistics = libraryStatisticsMap.get(library);
+            final IntHistogram.CDF templateSizeCDF = libraryStatistics.getCDF();
+            /*
+            final IntHistogram.CDF templateSizeCDF = libraryStatisticsMap.get(
+                    ((BreakpointEvidence.ReadEvidence) evidence).getReadGroup()
+            ).getCDF();
+            */
+            final int cdfBin = Integer.min(Math.abs(((BreakpointEvidence.ReadEvidence) evidence).getTemplateSize()),
+                                           templateSizeCDF.size() - 1);
+            return templateSizeCDF.getFraction(cdfBin);
+        } else if(evidence instanceof BreakpointEvidence.TemplateSizeAnomaly) {
+            // For TemplateSizeAnomaly, return readCount scaled by coverage
+            return (double)(((BreakpointEvidence.TemplateSizeAnomaly) evidence).getReadCount()) / coverage;
```

Comment:
I renamed the feature to templateSizeOrReadCount to make it a bit more clear that it's either-or. The alternative is to split into two feature columns and carry around the equivalent of an extra NaN for everything. I think that will make training the classifier (slightly) slower or less effective, so I'd like to avoid that if possible.

Reply:
Doesn't this possibly allow the classifier to do something silly like cut the data based on this field without regards to evidence type, which doesn't really make any sense? I would think that forcing the classifier to learn that the field means different things for different evidence types would make it less effective, not more, than explicitly telling it that template size anomalies don't have template sizes and read based evidence doesn't have read counts?